data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/BackoffUtils.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Random;
+
+public class BackoffUtils {
```

Comment:
We do need a different mechanism in place for the entire thread. As the SQS processing runs in its own thread, it must not end until the the source is shutdown.

I think we should use both:
* AWS SDK retry to help with transient errors
* Catch exceptions in the thread, suppress, and wait. This goes beyond AWS SDK errors. Say, some message in the SQS queue is not actually JSON. We don't want the thread to fail for a single bad message.

That being said, perhaps this will not need any sophisticated back-off. Perhaps a straightforward `Thread.sleep()` is enough.